### PR TITLE
Add trace.id to indexed deprecation logs

### PR DESCRIPTION
**Pull request by pgomulka**
https://github.com/elastic/elasticsearch/pull/75331 missed adding
trace.id to indexed deprecation logs (these use ECSJsonLayout instead of
ESJsonLayout).

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample

File: server/src/main/java/org/elasticsearch/common/logging/TraceIdConverter.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public static String getTraceId() {
     public void format(LogEvent event, StringBuilder toAppendTo) {
         String traceId = getTraceId();
         if (traceId != null) {
-            toAppendTo.append("\"trace.id\": \"" + traceId + "\"");
+            toAppendTo.append(traceId);
```
Comment (grcevski) [inline server/src/main/java/org/elasticsearch/common/logging/TraceIdConverter.java:55]: What's the reason for this change? Is it so that we can reuse the traceId formatter in EcsJsonLayout.java? 

I wonder if this changes behaviour a bit, previously if traceId was null the format would not append anything, but now we might get "trace.id:null". Is that OK?

Comment (grcevski) [reply server/src/main/java/org/elasticsearch/common/logging/TraceIdConverter.java:55]: Maybe notEmpty is smart enough ? :)

Comment (pgomulka) [reply server/src/main/java/org/elasticsearch/common/logging/TraceIdConverter.java:55]: yes, notEmpty will check if any of the variables in its "body" are evaluated

**Review by grcevski — APPROVED** [review]
LGTM!
